### ISSUE TITLE
[📝 Docs]: Fix documentation anomalies [UI/UX change]

### ISSUE THREAD

**Issue by bhavukkalra** [json-schema-org/website#653]
### What Docs changes are you proposing?

on the page [getting-started-step-by-step](https://json-schema.org/learn/getting-started-step-by-step) we see a lot of anomalies in the numbering. For example - 

![image](https://github.com/json-schema-org/website/assets/41142068/a45bcd62-470a-42d1-82ec-83ba7405c01a)

and 

![image](https://github.com/json-schema-org/website/assets/41142068/ce1b92ae-5892-40e9-8a0d-a295018a4447)

CC - @benjagm  @gregsdennis 
Do let me know if they seem valid to  be fixes and I can go ahead and create a PR for the same?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bhavukkalra** [comment]
On a similar line of thought. Adding `required` keys "productId", "productName" was kinda confusing. The docs does mentions that they are required but it can't be observed in the changes made. 
It is only some time later that we see the line of code

```json
"required": [ "productId", "productName", "price]" 
```

Where it starts to make sense. 

I think there is a better way to write it like this
 - Add the required keyword product id by appending this 
```json

  "properties": {
    "productId": {
      "description": "The unique identifier for a product",
      "type": "integer"
    }
  },

  "required": ["productId"]

```

TLDR - Update the `required` key as we go along adding new attributes. Not all at once

**Comment by benjagm** [comment]
Thanks for sharing your findings. I think this issue should be divided into 2: one for the list numbering and another for the rest of comments regarding formatting, wording and snippets

**Comment by bhavukkalra** [comment]
Sure. Thanks for your inputs @benjagm. Created a new issue https://github.com/json-schema-org/website/issues/658 for the snippets, wording change. In process of creating a PR for both soon.

**Comment by benjagm** [comment]
Once you have the issues created that way let's continue the discussion to complete the triage to move the issue to available status. That is the point when you can create the PR.

**Comment by gregsdennis** [comment]
I think the issue here is alignment of the code blocks.  They should be indented so that the block appears as part of the numbering.

For example:

```
1. Some text here
    ```
    and then an indented code block
    ```
2. Another text block
```

Versus...

````
1. Some text here
```
and then a code block
```
2. Another text block
````

(Github seems to render the numbering correctly in both, but I'm not sure the website does.)

**Comment by benjagm** [comment]
This is how I fixed the the problem with the repeated formats:

https://github.com/json-schema-org/website/commit/8001fe3b6e6fb43a985104677d989ae29b3c26c3

**Comment by bhavukkalra** [comment]
Hey Thanks for that benjamin. It was a markdown issue after all. Found the root cause over here. [markdown-continue-numbered-list](https://stackoverflow.com/questions/18088955/markdown-continue-numbered-list).

It seems like markdown uses spacing between ordered lists pairs to determine if a code block is still inside the scope or not. 
But yeah, We'll need to see how many more pages are there where this fix needs to be applied.
